pkg/network/config/config.go

```diff
@@ -270,22 +263,22 @@ func New() *Config {
 		ServiceMonitoringEnabled: cfg.GetBool(join(smNS, "enabled")),
 		DataStreamsEnabled:       cfg.GetBool(join(dsNS, "enabled")),
 
-		CollectTCPv4Conns: true,
-		CollectTCPv6Conns: true,
+		CollectTCPv4Conns: cfg.GetBool(join(netNS, "collect_tcp_v4")),
+		CollectTCPv6Conns: cfg.GetBool(join(netNS, "collect_tcp_v6")),
 		TCPConnTimeout:    2 * time.Minute,
 		TCPClosedTimeout:  1 * time.Second,
 
-		CollectUDPv4Conns: true,
-		CollectUDPv6Conns: true,
+		CollectUDPv4Conns: cfg.GetBool(join(netNS, "collect_udp_v4")),
+		CollectUDPv6Conns: cfg.GetBool(join(netNS, "collect_udp_v6")),
 		UDPConnTimeout:    defaultUDPTimeoutSeconds * time.Second,
 		UDPStreamTimeout:  defaultUDPStreamTimeoutSeconds * time.Second,
 
 		OffsetGuessThreshold:           uint64(cfg.GetInt64(join(spNS, "offset_guess_threshold"))),
 		ExcludedSourceConnections:      cfg.GetStringMapStringSlice(join(spNS, "source_excludes")),
 		ExcludedDestinationConnections: cfg.GetStringMapStringSlice(join(spNS, "dest_excludes")),
 
-		MaxTrackedConnections:          uint(cfg.GetInt(join(spNS, "max_tracked_connections"))),
-		MaxClosedConnectionsBuffered:   cfg.GetInt(join(spNS, "max_closed_connections_buffered")),
+		MaxTrackedConnections:          uint32(cfg.GetInt64(join(spNS, "max_tracked_connections"))),
```

Comment:
Why read this as an `int64` to begin with?

Reply:
So it can handle the full range of `uint32`. Our version of viper doesn't currently support uint, but I'm asking to get it added.